### PR TITLE
badger: add manual flatten to remove old version keys

### LINKED ISSUE #114: ngm data directory is very large
## Background

In a TiDB cluster which have `pd*3`, `tidb*8`, `tikv*8`, `2*tiflash`, and running with ng-monitoring more than 1 month.

## Problem

The `docdb` data directory of `ng-monitoring` is very large, it's not expected, and change the `data_retention_seconds` from `5 day` to `3 day` doesn't release disk space after 1 day.

```shell
root@436 data]# du -sh docdb/
1.2T    docdb/
```

Then, I write a script to scan the total badger db key/value size, only have `223GB`.

Then, I manually call badger `Flatten` API to do force compaction, then after 1 day( wait for gc ), the `docdb` data directory already release the disk space.

```shell
root@436 data]# du -sh docdb/
314G    docdb/
```

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #114

### What is changed and how it works?

Call badger `Flatten` API regularly.

`Flatten` uses to remove the old version keys, otherwise, the value log gc won't release the disk space which occupied
 by the old version keys.

### CODE REVIEW

File: database/document/gc.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	if !needFlatten(db) {
+		return
+	}
+	err := db.Flatten(runtime.NumCPU()/2 + 1)
+	if err != nil {
+		log.Error("badger flatten failed", zap.Error(err))
+		return
+	}
+	ts := time.Now().Unix()
+	err = storeLastFlattenTs(db, time.Now().Unix())
```
Comment (zhongzc) [inline database/document/gc.go:81]:
```suggestion
	err = storeLastFlattenTs(db, ts)
```

File: database/document/gc.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+)
+
+var (
```
Comment (zhongzc) [inline database/document/gc.go:13]: const?

File: database/document/gc.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+func runValueLogGC(db *badger.DB) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			log.Error("panic when run badger value log",
+				zap.Reflect("r", r),
+				zap.Stack("stack trace"))
+		}
+	}()
```
Comment (zhongzc) [inline database/document/gc.go:51]: Why only value log gc needs recovery?

File: database/document/gc.go (hunk 2 above)
Comment (crazycs520) [inline database/document/gc.go:13]:
```suggestion
const (
```

File: database/document/gc.go (hunk 3 above)
Comment (crazycs520) [reply database/document/gc.go]: already move this recover to `runGC` function.

File: database/document/gc.go (hunk 2 above)
Comment (crazycs520) [reply database/document/gc.go:13]: no, it can't be const.

**Review by zhongzc — APPROVED** [review]

**Pull request merged**